source/common/common/logger.h

```diff
@@ -517,6 +522,25 @@ class EscapeMessageJsonString : public spdlog::custom_flag_formatter {
  */
 #define ENVOY_LOG(LEVEL, ...) ENVOY_LOG_TO_LOGGER(ENVOY_LOGGER(), LEVEL, ##__VA_ARGS__)
 
+#define ENVOY_TAGGED_LOG_TO_LOGGER(LOGGER, LEVEL, TAGS, FORMAT, ...)                               \
+  do {                                                                                             \
+    if (ENVOY_LOG_COMP_LEVEL(LOGGER, LEVEL)) {                                                     \
```

Comment:
@mattklein123 I added this check for the logger level before doing the serialization and proceeding with the costlier operations - are there more checks I have missed that should be added?

Reply:
In your example you construct the map outside of the macro. We would optimally like to construct the map inside the macro if possible. Can you try that out?

Reply:
To make sure I understand you correctly - do you mean to move the code of ``serializeLogTags`` to inside the macro?

Reply:
What I mean is I could imagine doing something like:

```
ENVOY_TAGGED_LOG(info, {make tags here somehow}, "some_message {}", "val");
```

I just want to make sure that works correctly also.

Reply:
Got it. I think the right thing to do would replace the map with ``std::vector<std::pair<std::string, std::string>>``, since we don't do lookups in the map anyways. The vector will allow both pre-created values as well as inline vectors.
I'll wrap this up along with unit tests in next iteration.

Reply:
Best option I could find is wrapping the tags map in parentheses, for example:
```
ENVOY_TAGGED_LOG(info, (std::map<std::string, std::string>{{"key", "val"}}), "some_message {}", "val");
```

If you define some ``using`` for example ``using LogTagsMap = std::map<std::string, std::string>`` then it would look like ``ENVOY_TAGGED_LOG(info, (LogTagsMap{{"key", "val"}}), "some_message {}", "val");``, which is a bit nicer.

It's a bit awkward, but it works.. wdyt?